### ISSUE TITLE
Closed Blocks not conserved after leaving the page opened after that

### ISSUE THREAD

**Issue by RuochenFu21** [kiwix/kiwix-apple#737]
https://github.com/kiwix/kiwix-apple/assets/62882800/e1a7a43c-2fa4-4e5b-b899-a5aff8d1e755

**Comment by kelson42** [comment]
This is an issue for `openzim/mwoffliner`. I will move the issue.

**Comment by kelson42** [comment]
Kamino closed and cloned this issue to [openzim/mwoffliner](https://github.com/openzim/mwoffliner/issues/2281)

**Comment by benoit74** [comment]
To me, this is a reader issue as discussed in https://github.com/openzim/mwoffliner/issues/2281, let's keep both issue open until we've settled on this discussion

**Comment by kelson42** [comment]
@BPerlakiH @benoit74 On my side, I'm OK to consider this issue as a reader issue but... I'm not sure to know - at a software designe level - how the solution should looks like. Do you mean it would be possible to somehow remember the execution context to remember exactly the state of the HTML/CSS/JS runtime engine?

**Comment by benoit74** [comment]
Maybe there is simply no solution. I don't know much about how the "web document" is rendered at kiwix-apple level besides that it uses `WKWebView` component unless I'm mistaken. Probably Apple considers that it has the right to destroy this component "at will" when app is in background. Only thing to do would then be ensure we do not recreate a `WKWebView` while we could have reused one, but nothing else is possible. And I still don't think we should invest in some JS code to store state of the page in "browser". So probably no solution to this problem at all.

**Comment by BPerlakiH** [comment]
I did the following investigation on this:

## Double checking Safari browser on iOS and macOS

1) History and scroll position

Safari "remembers" the browsing history and the last scroll position and saves it even if the app is killed and restarted.

2) Collapsible

Safari remembers the collapsible state of html elements within the same app sessions throughout the navigation history. Eg. going back to a former page, the collapsible elements remain as they were before.
However it won't remember them if the app is restarted.
For testing I've used: https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_collapsible_symbol

3) Kiwix App:
From my testing on both iPhone and macOS with the current release it seems to behave correctly and exactly the same way as Safari browser. As of the latest main version:

https://github.com/user-attachments/assets/e669dea3-e2f5-4f60-9905-43c95d662075

I did tested it on a real device as well, but the recording from the simulator shows the touch interactions better.

Based on the above, I am closing this issue.

**Comment by furuochen-dev** [comment]
That’s rather strange. I’m currently on 3.9.0 build 227 the newest version, neither the location in page or the collapsed sections seems to be conserved.

https://github.com/user-attachments/assets/d1240589-5d7a-4587-ab91-0a41034cebca

**Comment by kelson42** [comment]
@BPerlakiH do you know why it behaves differently zhan in your test?

**Comment by BPerlakiH** [comment]
@kelson42 The only reason for a different behavior will be on the content side.
@RuochenFu21 can you send the download link to the ZIM file you are using please?

**Comment by kelson42** [comment]
We should not have any changes by switching tabs... never.

**Comment by BPerlakiH** [comment]
I need to have a look at the content myself locally. These reported videos without "tap indicator" are not something I am capable of following to be honest.
@RuochenFu21 please provide a download link to the ZIM file you are using.

**Comment by furuochen-dev** [comment]
I hope this screenshot helps.

![Image](https://github.com/user-attachments/assets/e819fd0e-7ffc-47a6-9bad-4d791c54f43f)

**Comment by BPerlakiH** [comment]
Thank you @RuochenFu21 !

I've downloaded this ZIM:
``wiktionary_en_all_nopic_2024-05.zim``

and I could reproduce this issue a 100%.

It is still a content issue. There's either something was already broken on the site when it was "zimified" or something is missing from the ZIM file.
On the debug console I can see the following JS errors:

<img width="1552" alt="Image" src="https://github.com/user-attachments/assets/9cd74fe1-8034-4df5-9e02-2a137af2bacb" />
<img width="1440" alt="Image" src="https://github.com/user-attachments/assets/611448ff-045f-461a-b9b4-10aeb2ab0130" />

From this it seems that the toggling of those collapsible elements gets stuck.
Although it looks like a standard JS extension from wikipedia, see here: https://doc.wikimedia.org/mediawiki-core/master/js/Hooks.html

By the first look, it seems like a timing issue in the loading, as on other breakpoints the a similarly starting code works.

**It's not something we can fix on the reader side.**

Luckily the new website no longer uses these collapsible elements, see eg here:
https://en.wiktionary.org/wiki/humor

So once the ZIM file will be recreated with the new content, it should solve this issue.

**Comment by kelson42** [comment]
Conclusion: Kiwix basicaly works fine if the js is correct but here, with this Wiktionary ZIM file from 2024, it seems the js was not correct. Meanwhil, new ZIM files of Wiktionary have been fixed and don't suffer anymore of this kind of issue. There is nothing more to do either on Kiwix of MWoffliner (the scraper), therefore closing issue.

**Comment by furuochen-dev** [comment]
Now, well, it’s just no longer collapsible, when it is in the real website, is this expected behavior?

<img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/0ee26ee1-6325-4edf-9a02-97d2c91216fe" />

**Comment by kelson42** [comment]
@RuochenFu21 Yes we won't provide anymore collapsible section for content issue from MediaWiki.
